File: LeetCodeSolutions/LeetCode_0048.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        l = len(matrix)
        for i in range(l):
            for j in range(i, l):
                if i != j:
                    # tmp = matrix[i][j]
                    matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
                    # matrix[j][i] = tmp
        for i in range(l):
            for j in range(l // 2):
                matrix[i][j], matrix[i][l - 1 - j] = matrix[i][l - 1 - j], matrix[i][j]

File: LeetCodeSolutions/LeetCode_0048.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        if (matrix.empty()) return;
        int row = matrix.size(), col = matrix[0].size();
        for(int r = 0; r < row / 2; ++r){
            for(int c = 0; c < col; ++c) swap(matrix[r][c], matrix[row -  r - 1][c]);
        }
        for (int r = 0; r < row; ++r)
            for(int c = r + 1; c < col; ++c){
                swap(matrix[r][c], matrix[c][r]);
            }
        return;
    }
};